src/conversational-search-interactions-api.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+const RESPONSE_SERVER_ERROR = require('./api').RESPONSE_SERVER_ERROR;
+require('es6-promise').polyfill();
+
+const conversationalSearchInteractionsInstance =
+  require('./api').conversationalSearchInteractionsInstance;
+
+var putSentimentClick = function (apiHostname, sitekey, conversationId, sentimentValue) {
+  return new Promise((resolve, reject) => {
+    conversationalSearchInteractionsInstance
+      .put(`https://${apiHostname}/v2/indices/${sitekey}/conversations/${conversationId}/rating`, {
```

Comment:
I think sending small analytics data should be done with [sendBeacon](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/sendBeacon). Backend endpoint will need to use POST for beacon.

Another point to clarify: how does backend handle this case when user clicks thumbs-up -> un-click -> then click again? Does the whole thing count as 1 or 2 "thumbs-up"? If 1, I'd suggest sending `sessionId` so backend can handle the calculating logic.

Check out this endpoint https://api.addsearch.com/v1/stats/2c32bf3eb06b30af5f8208481aea3e8b/ when searching in https://demo.addsearch.com/addsearch-ready-made-solutions/widget.html

Reply:
Sentiment is more like a feature of converstational search. Maybe in the end we decide to send a stats event as well in parallel. But I think at it's core it is a direct interaction with the feature.

As to the value sent: The frontend decides which value to pass to putSentimentClick. If nothing is selected or the other value is selected at the moment the respective value is sent, if something is selected that was already selected the value becomes "neutral" i.e. un-select.

Reply:
Makes sense, I'll approve this PR.